lib/pages/widget_popup.dart

```diff
@@ -0,0 +1,301 @@
+              SizedBox(
+                height: 5,
+              ),
+              DropdownButtonFormField<String>(
+                hint: Text(
+                  getString(context, 'item_type'),
+                ),
+                onChanged: (text) {
+                  setState(() {
+                    _type = text;
+                  });
+                },
+                value: _type,
+                items: <String>[
+                  getString(context, 'item_unit'),
+                  getString(context, 'item_liters'),
+                  getString(context, 'item_grams'),
+                  getString(context, 'item_kilos')
+                ].map((String value) {
+                  return new DropdownMenuItem<String>(
+                    value: value,
+                    child: new Text(value),
+                  );
+                }).toList(),
+              ),
+              SizedBox(
+                height: 5,
+              ),
+              RaisedButton(
+                onPressed: () {
+                  if (itemNameController == null ||
+                      itemCountController == null ||
+                      _type == null ||
+                      itemNameController.text == "" ||
+                      itemCountController.text == "") {
+                    setState(() {
+                      alert = getString(context, "popup_alert");
+                    });
+                  } else {
+                    if (buttonName == getString(context, "popup_update")) {
+                      uddapteItemInList();
+                    } else {
+                      addItemToList();
+                    }
+                  }
+                },
+                child: Text(
+                  buttonName,
+                  style: TextStyle(
+                      color: Colors.grey[600], fontWeight: FontWeight.bold),
+                ),
+                color: WHITE,
+              ),
+              Text(
+                alert,
+                style:
+                    TextStyle(color: Colors.red, fontWeight: FontWeight.bold),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  void clearPopupFields() {
+    itemCountController.clear();
+    itemNameController.clear();
+    _name = null;
+    _count = 0;
+    _type = null;
+  }
+
+  void uddapteItemInList() async {
+    await databaseReference.collection("items").document(listUuid).updateData({
+      itemUuid: {
+        'label': _name,
+        'quantity': _count,
+        'unit': _type,
+      }
+    });
+    Navigator.of(context).pop();
+    clearPopupFields();
+  }
+
+  void addItemToList() async {
+    var uuid = Uuid();
+    var newUuid = uuid.v4();
+    bool doesDocumentExist = false;
+
+    await Firestore.instance
+        .collection("items")
+        .document(listUuid)
+        .get()
+        .then((value) {
+      doesDocumentExist = value.exists;
+    });
+
+    //Create the item
+    if (doesDocumentExist == true) {
```

Comment:
doesDocumentExist will always be false. You call addItemToList only if you are not in update mode so it is not necessary to check if document exists. Just add it